src/tgp.cpp

```diff
@@ -673,7 +686,7 @@ static void HeightMapAdjustWaterLevel(Amplitude water_percent, Height h_max_new)
 	hist = HeightMapMakeHistogram(h_min, h_max, hist_buf.data());
 
 	/* How many water tiles do we want? */
-	desired_water_tiles = A2I(((int64_t)water_percent) * (int64_t)(_height_map.size_x * _height_map.size_y));
+	desired_water_tiles = water_percent * (int64_t)(_height_map.size_x * _height_map.size_y) / 1024;
```

Comment:
```suggestion
	desired_water_tiles = water_percent * _height_map.size_x * _height_map.size_y / 1024;
```
Less casting is better. The `water_percent` already being `int64_t` makes the rest `int64_t` as well.